sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -68,10 +68,10 @@ namespace Azure { namespace Core { namespace Http {
      */
     virtual void Rewind()
     {
-      // Exception is not meant to be caught. This is rather an indication of a bad program that
-      // needs to be updated. This is the reason of using `throw` alone.
-      throw "The upload BodyStream doesn't support Rewind which is required to guarantee fault "
-            "tolerance when retrying the operation.";
+      throw std::runtime_error(
```

Comment:
std::invalid_argument makes more sense.  Its a logic error so should make sense for folks to handle that at dev time

Reply:
Imagine the user calls `stream.Rewind()` and get's back `invalid_argument`. How would they reason about that? They didn't pass in any arguments.

It feels strange to me we'd throw such an exception, even if it is something the caller needs to handle at dev time.

Reply:
runtime_error exceptions are for
```
std::runtime_error
Defines a type of object to be thrown as exception. It reports errors that are due to events beyond the scope of the program and can not be easily predicted.
```

In this particular case the user did something in their code that can be predicted and thus logic_error is the type of exception we should use
```
std::logic_error
Defines a type of object to be thrown as exception. It reports errors that are a consequence of faulty logic within the program such as violating logical preconditions or class invariants and may be preventable.
```